HunterPie.Integrations/Datasources/MonsterHunterRise/Entity/Player/MHRPlayer.cs

```diff
@@ -434,7 +434,10 @@ private void GetPlayerDebuffAbnormalities()
             MHRDebuffStructure abnormality = new();
 
             // Only read memory if the required sub Id is the required one for this abnormality
-            if (abnormSubId == schema.WithValue)
+            // For Leeched debuff, Compares whether "WithValue" is negative or positive
+            if (schema.IsInfinite)
+                abnormality.Timer = (schema.WithValue >= 0 && abnormSubId == schema.WithValue) || (schema.WithValue < 0 && abnormSubId != schema.WithValue) ? AbnormalityData.TIMER_MULTIPLIER : 0;
+            else if ((schema.WithValue >= 0 && abnormSubId == schema.WithValue) || (schema.WithValue < 0 && abnormSubId != schema.WithValue))
```

Comment:
Honestly, I took a look at the Leeched debuff to understand what exactly you're trying to do here and I understand now but I don't think this is a good solution, the Abnormality scanners should be agnostic to what abnormality they're scanning, that's the whole point of why I keep all abnormalities definition in a xml file with a well-defined behavior schema.

This solution is making it so it knows the existence of Leeched and how it behaves, which goes against the idea of the abstraction, and not only that, it's also breaking the `WithValue` definition when the abnormality is also marked as `IsInfinite`, making it behave as a negation of what `WithValue` stands for.

I'd rather just create a new field of `WithValueNot` than break the `WithValue` definition.

Reply:
For that you'll have to update the `AbnormalitySchema`, the `AbnormalityData` parser, probably add an enum for the schema so the scanners should look at the `WithValue` or `WithValueNot` and process the right operation. Also, an abnormality should **NOT** have both `WithValue` and `WithValueNot` defined, only one of them.

Reply:
The first review was about the code legibility, I didn't really look at the solution because that many if/elses was making it hard to read the code. Now the review is about the solution itself.

Reply:
oh, sorry. What I said that not pushed.
I gave up because I think there were a lot of source changes.

Reply:
I added enum and edited parser and scan method.